### PR TITLE
Advanced Usage Edits (cnoon)

### PR DESCRIPTION
**Pull request by cnoon**
Small tweaks and edits on the Advanced Usage docs.

### CODE REVIEW

File: Documentation/AdvancedUsage.md [hunk 1]
```diff
@@ -218,14 +221,14 @@ Each request performed by Alamofire is encapsulated by particular class, `DataRe
...
 4. `Session` calls the `Request` to create the `URLSessionTask` to perform the network request based on the `URLRequest`.
 5. Once the `URLSessionTask` is complete and `URLSessionTaskMetrics` have been gathered, the `Request` executes its `Validator`s. 
 6. Request executes any response handlers, such as `responseDecodable`, that have been appended.
 
-At any one of these steps, a failure can be indicated through a created or received `Error` value, which is then passed to the associated `Request`. For example, aside from steps 1 and 4, all of the steps above can create an `Error` which is then passed to the response handlers or available for retry.
+At any one of these steps, a failure can be indicated through a created or received `Error` value, which is then passed to the associated `Request`. For example, aside from steps 1 and 4, all of the steps above can create an `Error` which is then passed to the response handlers or available for retry. Here are a few examples of what can or cannot fail throughout the `Request` pipeline.
```
Comment (cnoon) [inline Documentation/AdvancedUsage.md]: This paragraph didn't lead into the list below it very well. I tried to improve it, but I wasn't entirely sure if this is what you intended @jshier. Let me know.

File: Documentation/AdvancedUsage.md [hunk 2]
```diff
@@ -522,13 +532,14 @@ let manager = ServerTrustManager(evaluators: serverTrustPolicies)
...
 This `ServerTrustManager` will have the following behaviors:
+
 - `cert.example.com` will always use certificate pinning with default and host validation enabled , thus requiring the following criteria to be met in order to allow the TLS handshake to succeed:
 	- Certificate chain *must* be valid.
 	- Certificate chain *must* include one of the pinned certificates.
 	- Challenge host *must* match the host in the certificate chain's leaf certificate.
 - `keys.example.com` will always use public key pinning with default and host validation enabled, thus requiring the following criteria to be met in order to allow the TLS handshake to succeed:
 	- Certificate chain *must* be valid.
-	- Leaf certificates *must* include one of the pinned public keys.
+	- Certificate chain *must* include one of the pinned public keys.
```
Comment (cnoon) [inline Documentation/AdvancedUsage.md]: It evaluates the entire chain for both certs and public keys looking for a match.

File: Documentation/AdvancedUsage.md [hunk 3]
```diff
@@ -653,7 +665,12 @@ let session = Session(eventMonitors: [logger])
 ```
 
 ## Making Requests
-As a framework, Alamofire has two main goals: to enable the easy implementation of network requests for prototypes and tools, and to serve as the generic foundation of app networking. It accomplishes these goals through the use of powerful abstractions, providing useful defaults, and included implementations of common tasks. However, once use of Alamofire  has gone beyond a few requests, it’s necessary to move beyond the high level, default implementations into behavior customized for particular applications. Alamofire provides the `URLConvertible` and `URLRequestConvertible` protocols to help with this customization.
+As a framework, Alamofire has two main goals: 
+
+1. To enable the easy implementation of network requests for prototypes and tools
+2. To serve as the generic foundation of app networking
```
Comment (cnoon) [inline Documentation/AdvancedUsage.md]: I thought we could put a finer point on it by actually listing them out.

Comment (jshier) [reply Documentation/AdvancedUsage.md]: Looks good. Do these need periods?

File: Documentation/AdvancedUsage.md (hunk 2 above)
Comment (jshier) [reply Documentation/AdvancedUsage.md]: 👍

File: Documentation/AdvancedUsage.md (hunk 1 above)
Comment (jshier) [reply Documentation/AdvancedUsage.md]: I think it's an improvement. If users want even more detail we'll have to come up with a better way to express it.

**Pull request merged**